numpy/lib/_iotools.py

```diff
@@ -18,18 +18,19 @@ def _decode_line(line, encoding=None):
     ----------
     line : str or bytes
          Line to be decoded.
+    encoding : str
+         Encoding used to decode `line`.
 
     Returns
     -------
     decoded_line : unicode
          Unicode in Python 2, a str (unicode) in Python 3.
 
     """
+    if encoding is None:
+        encoding = 'latin1'
     if type(line) is bytes:
-        if encoding is None:
-            line = line.decode('latin1')
-        else:
-            line = line.decode(encoding)
```

Comment:
Could you move the changed stanza back under the `if` so it would become
```
if type(line) is bytes:
    if not encoding:
                encoding = 'latin1'
    line = line.decode(encoding)
```